cnd/src/db/mod.rs

```diff
@@ -0,0 +1,173 @@
+mod models;
+mod schema;
+
+pub use crate::db::models::Metadata;
+
+use crate::{
+    db::models::NewMetadata,
+    swap_protocols::{
+        metadata_store::{self, AssetKind, Error, LedgerKind, MetadataStore, Role},
+        SwapId,
+    },
+};
+use diesel::{self, prelude::*, sqlite::SqliteConnection};
+use libp2p::PeerId;
+use migrations_internals;
+use std::{
+    convert::TryFrom,
+    fs::File,
+    io::stdout,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+/// This module provides an Sqlite backed persistent MatadataStore.
+
+#[derive(Debug)]
+pub struct SqliteMetadataStore {
+    db: PathBuf,
+}
+
+impl SqliteMetadataStore {
+    /// Return a handle that can be used to access the database.
+    ///
+    /// When this returns an Sqlite database exists at 'db' and connection to
+    /// the database has been verified.
+    pub fn new(db: Option<PathBuf>) -> Result<SqliteMetadataStore, Error> {
+        let db = db
+            .or(default_db_path())
+            .ok_or_else(|| Error::Path("Failed to get database path".to_string()))?;
+
+        create_database_if_needed(&db)?;
+        log::info!("Sqlite database file: {}", db.display());
+
+        let migrations = default_migrations_dir()
+            .ok_or_else(|| Error::Path("Failed to get migrations directory".to_string()))?;
+
+        create_migrations_directory_if_needed(&migrations)?;
+        run_migrations(&db, &migrations)?;
```

Comment:
I'm no database expert but I think your concerns are valid if/when we add a second migration.  Currently we have only a single migration so it can be run automatically.  I think a comment it this point in the code to note this could be a good idea.

Reply:
> Do we really want to automatically run migration without a warning?

In the spirit of "it just works TM" I would say yes.

> The user should then have created a back up prior running the migration.

Cnd should probably create a backup of the database automatically before running migrations.

I would not want to burden the user with that. That seems like an unnecessary complication if the software is telling you to do stuff manually that it just could have done automatically.